src/citrine/exceptions.py

```diff
@@ -84,6 +85,21 @@ class WorkflowNotReadyException(RetryableException):
     pass
 
 
+class PollingTimeoutError(NonRetryableException):
+    """Polling for an asynchronous result has exceeded the timeout."""
+
+    pass
+
+
+class JobFailureError(NonRetryableException):
+    """The asynchronous job completed with the given failure message."""
+
+    def __init__(self, message: str, *, job_id: UUID, failure_reasons: List[str]):
```

Comment:
This doesn't make a ton of sense.  If we have 3 required arguments, then I think they should all be keyword only.

Also, I think this is the source of the test failures.  job_id and failure_reasons are keyword only but the constructor invocation [here](https://github.com/CitrineInformatics/citrine-python/pull/479/files#diff-f6170f32137e1e86f98504d21a90b363c2e46a166001311cdd69a497d54610beR290) isn't using keywords

Reply:
Agreed. Remove that.